pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ServerCnxTest.java

```diff
@@ -2307,4 +2308,21 @@ public void handleConnectWithServiceNotReady() throws Exception {
         assertEquals(error.getError(), ServerError.ServiceNotReady);
         channel.finish();
     }
+
+    @Test(timeOut = 30000)
+    public void handlePartitionMetadataRequestWithServiceNotReady() throws Exception {
+        resetChannel();
+        setChannelConnected();
+        CompletableFuture<Boolean> successFuture = CompletableFuture.completedFuture(true);
+        doReturn(successFuture).when(namespaceService).checkTopicExists(any(TopicName.class));
```

Comment:
Is it possible to remove this line2316 to line2317？

Reply:
No. It will make the test case more complex. I think it's not worth it .